clipper_admin/clipper_admin/docker/docker_container_manager.py

```diff
@@ -169,17 +218,55 @@ def start_clipper(self,
             query_container_id)
         run_query_frontend_metric_image(
             query_frontend_metric_name, self.docker_client, query_name,
-            self.common_labels, self.extra_container_kwargs)
-        setup_metric_config(query_frontend_metric_name,
+            frontend_exporter_image, self.common_labels,
+            self.extra_container_kwargs)
+
+        self.prom_config_path = tempfile.NamedTemporaryFile(
+            'w', suffix='.yml', delete=False).name
+        self.prom_config_path = os.path.realpath(
+            self.prom_config_path)  # resolve symlink
+        self.logger.info("Metric Configuration Saved at {path}".format(
+            path=self.prom_config_path))
+        setup_metric_config(query_frontend_metric_name, self.prom_config_path,
                             CLIPPER_INTERNAL_METRIC_PORT)
-        run_metric_image(self.docker_client, self.common_labels,
-                         self.prometheus_port, self.extra_container_kwargs)
+
+        self.prometheus_port = find_unbound_port(self.prometheus_port)
+        metric_labels = self.common_labels.copy()
+        metric_labels[CLIPPER_DOCKER_PORT_LABELS['metric']] = str(
+            self.prometheus_port)
+        metric_labels[CLIPPER_METRIC_CONFIG_LABEL] = self.prom_config_path
+        run_metric_image(self.docker_client, metric_labels,
+                         self.prometheus_port, self.prom_config_path,
+                         self.extra_container_kwargs)
 
         self.connect()
 
     def connect(self):
-        # No extra connection steps to take on connection
-        return
+        """
+        Use the cluster name to update ports. Because they might not match as in
+        start_clipper the ports might be changed.
+        :return: None
+        """
+        containers = self.docker_client.containers.list(
+            filters={
+                'label': [
+                    '{key}={val}'.format(
```

Comment:
@dcrankshaw the literal label is fixed here (same above). I don't know why github is still showing the old diff